enumerable.rb

```diff
@@ -0,0 +1,169 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_each
+    if block_given?
+      v = self
+      n = v.length
+      n.times do |i|
+        yield(v[i])
+      end
+    else
+      to_enum(:my_each)
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      v = self
+      n = v.length
+      n.times do |i|
+        yield(self[i], i)
+      end
+    else
+      to_enum(:my_each_with_index)
+    end
+  end
+
+  def my_select
+    value = self
+    if block_given?
+      temp = []
+      value.my_each do |a|
+        temp.push(a) if yield(a)
+      end
+      temp
+    else
+      to_enum(:my_select)
+    end
+  end
+
+  def my_all?(arg = nil)
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return false unless yield(a, b) }
+      else
+        value.my_each { |a| return false unless yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return false unless a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return false unless a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return false unless a }
+    end
+    true
+  end
+
+  def my_any?
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return true if yield(a, b) }
+      else
+        value.my_each { |a| return true if yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return true if a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return true if a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return true if a }
+    end
+    false
+  end
+
+  def my_none?
+    value = self
+    if block_given?
+      if value.is_a? Hash
+        value.my_each { |a, b| return false if yield(a, b) }
+      else
+        value.my_each { |a| return false if yield(a) }
+      end
+    elsif arg.is_a? Regexp
+      value.my_each { |a| return false if a =~ arg }
+    elsif arg.is_a? Class
+      value.my_each { |a| return false if a.class == arg }
+    elsif arg.nil?
+      value.my_each { |a| return false if a }
+    end
+    true
+  end
```

Comment:
Please, see comment on `my_all?` and `my_any?`

-----

[OPTIONAL]

I think if we could get the block from `my_none?` to `my_any?` we should only need something like `!my_any?` for `my_none?` to work fine. For example:

```ruby
  def my_none?(arg = nil, &block)
    !my_any?(arg = nil,&block)
  end
```

To make this work we have to use `block.call(element)` instead of `yield(element)` in `my_any?`